px-popover.html

```diff
@@ -270,7 +270,8 @@
                 if( this._isDescendant(this.$.popoverWrapper, event.target) ) {
                     return false;
                 } else {
-                    if(event.target.id === this.for) {
+                    if ((typeof (this.for) === 'string' && event.target.id === this.for) ||
+                        (typeof (this.for) === 'object' && event.target.id === this.for.id)) {
```

Comment:
If `this.for` is a reference to an HTMLElement, it does not necessarily need to have an ID. If that element does not have an ID, this check will fail: `event.target.id === this.for.id`. We might not want to do this at all, we might want to check for `this.for === event.currentTarget` in this case, see comment above.

Reply:
This is how the existing code works, I am not intending to fix the world here - just one specific bug. If `event.target.id` should not be used for the ID comparison then that change would affect the existing code-path as well and would require more thought and tests than are in the scope of this PR.